src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -190,7 +191,10 @@ private Type0CommandMessageSection<BsonDocument> CreateType0Section(ConnectionDe
             var dbElement = new BsonElement("$db", _databaseNamespace.DatabaseName);
             extraElements.Add(dbElement);
 
-            if (_readPreference != null && _readPreference != ReadPreference.Primary)
+            // if standalone, don't send read preference
```

Comment:
Not sure, but should this logic be also added to some other places like this:
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingQueryMessageWireProtocol.cs#L365
Should we also check places when this method is called: `QueryHelper.CreateReadPreferenceDocument`?

Reply:
The reason I didn't add a check in `QueryHelper.CreateReadPreferenceDocument` is because the check at [this line](https://github.com/ksadoff/mongo-csharp-driver/blob/16cdf44804719a981779424dcf81ed12f2b8a292/src/MongoDB.Driver.Core/Core/Operations/QueryHelper.cs#L57) is broad enough to include ServerType.Standalone (it creates a null readPreference for any ServerType that isn't ShardRouter). And that's why there's no check in `WrapCommandForQueryMessage`, since the logic is taken care of in QueryHelper.